File: Programação/Facul/aula pablo/java/portfolio3/concessionaria/GerenciadorVeiculos.java
package concessionaria;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import veiculos.CarroVenda;

public class GerenciadorVeiculos {
    private final List<CarroVenda> veiculos;

    public GerenciadorVeiculos() {
        this.veiculos = new ArrayList<>();
    }

    private void cadastrarVeiculo(Scanner scanner) {
        CarroVenda carro = new CarroVenda();

        System.out.print("Número de Série: ");
        carro.setNumeroSerie(scanner.nextLine());

        System.out.print("Marca: ");
        carro.setMarca(scanner.nextLine());

        System.out.print("Modelo: ");
        carro.setModelo(scanner.nextLine());

        System.out.print("Preço de Venda: ");
        carro.setPrecoVenda(new BigDecimal(scanner.nextLine()));

        veiculos.add(carro);
        System.out.println("\nVeículo cadastrado com sucesso.\n");
    }

    private void imprimirFicha(Scanner scanner) {
        if (veiculos.isEmpty()) {
            System.out.println("\nNenhum veículo cadastrado.\n");
            return;
        }

        System.out.print("\nNúmero de série do veículo: ");
        String numeroSerie = scanner.nextLine();

        veiculos.stream()
                .filter(v -> v.getNumeroSerie().equals(numeroSerie))
                .findFirst()
                .ifPresentOrElse(
                        v -> System.out.println("\n" + v.obterFicha() + "\n"),
                        () -> System.out.println("\nVeículo não encontrado.\n")
                );
    }

    private void exibirMenu() {
        System.out.println("1 -- Cadastrar Veículo para venda");
        System.out.println("2 -- Imprimir ficha do veículo para venda");
        System.out.println("3 -- Sair");
        System.out.print("Opção: ");
    }

    private void executar() {
        try (Scanner scanner = new Scanner(System.in)) {
            boolean executando = true;

            while (executando) {
                exibirMenu();
                String opcao = scanner.nextLine();

                switch (opcao) {
                    case "1" -> cadastrarVeiculo(scanner);
                    case "2" -> imprimirFicha(scanner);
                    case "3" -> executando = false;
                    default -> System.out.println("\nOpção inválida.\n");
                }
            }
        }
    }

    public static void main(String[] args) {
        new GerenciadorVeiculos().executar();
    }
}

File: Programação/Facul/aula pablo/c#/portfolio_3/concessionaria/GerenciadorVeiculos.cs
using Veiculos;

namespace Concessionaria;

public class GerenciadorVeiculos
{
    private readonly List<CarroVenda> _veiculos = [];

    private void CadastrarVeiculo()
    {
        var carro = new CarroVenda();

        Console.Write("Numero de Serie: ");
        carro.NumeroSerie = Console.ReadLine() ?? string.Empty;

        Console.Write("Marca: ");
        carro.Marca = Console.ReadLine() ?? string.Empty;

        Console.Write("Modelo: ");
        carro.Modelo = Console.ReadLine() ?? string.Empty;

        Console.Write("Preco de Venda: ");
        var textoPreco = Console.ReadLine();

        if (!decimal.TryParse(textoPreco, out var precoVenda))
        {
            Console.WriteLine("\nPreco invalido. Cadastro cancelado.\n");
            return;
        }

        carro.PrecoVenda = precoVenda;
        _veiculos.Add(carro);

        Console.WriteLine("\nVeiculo cadastrado com sucesso.\n");
    }

    private void ImprimirFicha()
    {
        if (_veiculos.Count == 0)
        {
            Console.WriteLine("\nNenhum veiculo cadastrado.\n");
            return;
        }

        Console.Write("\nNumero de serie do veiculo: ");
        var numeroSerie = Console.ReadLine() ?? string.Empty;

        var veiculo = _veiculos.FirstOrDefault(v => v.NumeroSerie == numeroSerie);

        if (veiculo is null)
        {
            Console.WriteLine("\nVeiculo nao encontrado.\n");
            return;
        }

        Console.WriteLine($"\n{veiculo.ObterFicha()}\n");
    }

    private static void ExibirMenu()
    {
        Console.WriteLine("1 -- Cadastrar Veiculo para venda");
        Console.WriteLine("2 -- Imprimir ficha do veiculo para venda");
        Console.WriteLine("3 -- Sair");
        Console.Write("Opcao: ");
    }

    private void Executar()
    {
        var executando = true;

        while (executando)
        {
            ExibirMenu();
            var opcao = Console.ReadLine();

            switch (opcao)
            {
                case "1":
                    CadastrarVeiculo();
                    break;
                case "2":
                    ImprimirFicha();
                    break;
                case "3":
                    executando = false;
                    break;
                default:
                    Console.WriteLine("\nOpcao invalida.\n");
                    break;
            }
        }
    }

    public static void Main(string[] args)
    {
        new GerenciadorVeiculos().Executar();
    }
}